src/irmin-pack/unix/dispatcher.ml

```diff
@@ -117,8 +120,16 @@ module Make (Fm : File_manager.S with module Io = Io.Unix) :
   let read_exn t ~off ~len ?volume_identifier buf =
     read_range_exn t ~off ~min_len:len ~max_len:len ?volume_identifier buf
 
+  let read_sparse_or_lower_exn t ~off ~max_len buf =
+    try Sparse.read_range_exn (get_prefix t) ~off ~min_len:1 ~max_len buf
```

Comment:
Any reason not to set `min_len=max_len`? Or do we expect it to be a true range read?
```suggestion
    try Sparse.read_range_exn (get_prefix t) ~off ~min_len:max_len ~max_len buf
```

Reply:
I would expect `min_len=max_len` to also work yes, since we shouldn't actually trigger read from both the prefix and the lower from the GC (we only hit the lower if there's no prefix). But `min_len:1` yields a cleaner interface where one doesn't need to know about that edge-case :P